### PR TITLE
修正ubuntu解析二维码输出到命令行窗口bash命令

### PR DESCRIPTION
**Pull request by luzihang123**
## qrencode -r 报错
```bash
qrencode: invalid option -- 'r'
Try `qrencode --help' for more information.
```

## qrencode --help
```
qrencode version 3.4.4
Copyright (C) 2006-2012 Kentaro Fukuchi
Usage: qrencode [OPTION]... [STRING]
Encode input data in a QR Code and save as a PNG or EPS image.

  -h, --help   display the help message. -h displays only the help of short
               options.

  -o FILENAME, --output=FILENAME
               write image to FILENAME. If '-' is specified, the result
               will be output to standard output. If -S is given, structured
               symbols are written to FILENAME-01.png, FILENAME-02.png, ...
               (suffix is removed from FILENAME, if specified)
  -s NUMBER, --size=NUMBER
               specify module size in dots (pixels). (default=3)

  -l {LMQH}, --level={LMQH}
               specify error correction level from L (lowest) to H (highest).
               (default=L)

  -v NUMBER, --symversion=NUMBER
               specify the version of the symbol. See SYMBOL VERSIONS for more
               information. (default=auto)

  -m NUMBER, --margin=NUMBER
               specify the width of the margins. (default=4 (2 for Micro QR)))

  -d NUMBER, --dpi=NUMBER
               specify the DPI of the generated PNG. (default=72)

  -t {PNG,EPS,SVG,ANSI,ANSI256,ASCII,ASCIIi,UTF8,ANSIUTF8}, --type={PNG,EPS,
               SVG,ANSI,ANSI256,ASCII,ASCIIi,UTF8,ANSIUTF8}
               specify the type of the generated image. (default=PNG)

  -S, --structured
               make structured symbols. Version must be specified.

  -k, --kanji  assume that the input text contains kanji (shift-jis).

  -c, --casesensitive
               encode lower-case alphabet characters in 8-bit mode. (default)

  -i, --ignorecase
               ignore case distinctions and use only upper-case characters.

  -8, --8bit   encode entire data in 8-bit mode. -k, -c and -i will be ignored.

      --rle    enable run-length encoding for SVG.

  -M, --micro  encode in a Micro QR Code. (experimental)

      --foreground=RRGGBB[AA]
      --background=RRGGBB[AA]
               specify foreground/background color in hexadecimal notation.
               6-digit (RGB) or 8-digit (RGBA) form are supported.
               Color output support available only in PNG and SVG.
  -V, --version
               display the version number and copyrights of the qrencode.

      --verbose
               display verbose information to stderr.

  [STRING]     input data. If it is not specified, data will be taken from
               standard input.

*SYMBOL VERSIONS
               The symbol versions of QR Code range from Version 1 to Version
               40. Each version has a different module configuration or number
               of modules, ranging from Version 1 (21 x 21 modules) up to
               Version 40 (177 x 177 modules). Each higher version number
               comprises 4 additional modules per side by default. See
               http://www.qrcode.com/en/about/version.html for a detailed
               version list.
```

### CODE REVIEW

**Comment by DalekFrog** [comment]
你这版本问题，我在ubuntu 2004 上安装的qrencode版本是4.0.2，没有问题，有-r 参数
```
qrencode version 4.0.2
Copyright (C) 2006-2017 Kentaro Fukuchi
Usage: qrencode [OPTION]... [STRING]
Encode input data in a QR Code and save as a PNG or EPS image.

  -h           display this message.
  --help       display the usage of long options.
  -o FILENAME  write image to FILENAME. If '-' is specified, the result
               will be output to standard output. If -S is given, structured
               symbols are written to FILENAME-01.png, FILENAME-02.png, ...
               (suffix is removed from FILENAME, if specified)
  -r FILENAME  read input data from FILENAME.
  -s NUMBER    specify module size in dots (pixels). (default=3)
  -l {LMQH}    specify error correction level from L (lowest) to H (highest).
               (default=L)
  -v NUMBER    specify the minimum version of the symbol. (default=auto)
  -m NUMBER    specify the width of the margins. (default=4 (2 for Micro))
  -d NUMBER    specify the DPI of the generated PNG. (default=72)
  -t {PNG,PNG32,EPS,SVG,XPM,ANSI,ANSI256,ASCII,ASCIIi,UTF8,ANSIUTF8}
               specify the type of the generated image. (default=PNG)
  -S           make structured symbols. Version must be specified.
  -k           assume that the input text contains kanji (shift-jis).
  -c           encode lower-case alphabet characters in 8-bit mode. (default)
  -i           ignore case distinctions and use only upper-case characters.
  -8           encode entire data in 8-bit mode. -k, -c and -i will be ignored.
  -M           encode in a Micro QR Code.
  -V           display the version number and copyrights of the qrencode.
  [STRING]     input data. If it is not specified, data will be taken from
               standard input.

  Try "qrencode --help" for more options.
```

File: README.md [hunk 1]
```diff
@@ -116,7 +116,7 @@
 
 ```bash
 $ sudo apt-get install qrencode zbar-tools # 安装二维码解析和生成的工具，用于读取二维码并在命令行输出。
-$ zbarimg qr_code.png > qrcode.txt && qrencode -r qrcode.txt -o - -t UTF8 # 解析二维码输出到命令行窗口。
+$ zbarimg qr_code.png > qrcode.txt && cat qrcode.txt | qrencode -o - -t UTF8 # 解析二维码输出到命令行窗口。
```
Comment (huanghyw) [inline README.md:119]: 可以可以，这种方式兼容性更好

**Review by huanghyw — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (weiqi88) [reply README.md:119]: 功能列表：                                                                                
 1.预约商品
 2.秒杀抢购商品
    
请选择:1
[W:pyppeteer.chromium_downloader] start chromium download.
Download may take a few minutes.

**Comment by luzihang123** [comment]
> 你这版本问题，我在ubuntu 2004 上安装的qrencode版本是4.0.2，没有问题，有-r 参数
> 
> ```
> qrencode version 4.0.2
> Copyright (C) 2006-2017 Kentaro Fukuchi
> Usage: qrencode [OPTION]... [STRING]
> Encode input data in a QR Code and save as a PNG or EPS image.
> 
>   -h           display this message.
>   --help       display the usage of long options.
>   -o FILENAME  write image to FILENAME. If '-' is specified, the result
>                will be output to standard output. If -S is given, structured
>                symbols are written to FILENAME-01.png, FILENAME-02.png, ...
>                (suffix is removed from FILENAME, if specified)
>   -r FILENAME  read input data from FILENAME.
>   -s NUMBER    specify module size in dots (pixels). (default=3)
>   -l {LMQH}    specify error correction level from L (lowest) to H (highest).
>                (default=L)
>   -v NUMBER    specify the minimum version of the symbol. (default=auto)
>   -m NUMBER    specify the width of the margins. (default=4 (2 for Micro))
>   -d NUMBER    specify the DPI of the generated PNG. (default=72)
>   -t {PNG,PNG32,EPS,SVG,XPM,ANSI,ANSI256,ASCII,ASCIIi,UTF8,ANSIUTF8}
>                specify the type of the generated image. (default=PNG)
>   -S           make structured symbols. Version must be specified.
>   -k           assume that the input text contains kanji (shift-jis).
>   -c           encode lower-case alphabet characters in 8-bit mode. (default)
>   -i           ignore case distinctions and use only upper-case characters.
>   -8           encode entire data in 8-bit mode. -k, -c and -i will be ignored.
>   -M           encode in a Micro QR Code.
>   -V           display the version number and copyrights of the qrencode.
>   [STRING]     input data. If it is not specified, data will be taken from
>                standard input.
> 
>   Try "qrencode --help" for more options.
> ```
对。
新买的服务器，没有及时更新。
及时更新，也可以避免这个版本的问题，方便命令行显示二维码
```bash
sudo apt-get update #更新源
sudo apt-get upgrade #更新已安装的包
```

**Comment by luzihang123** [comment]
> 你这版本问题，我在ubuntu 2004 上安装的qrencode版本是4.0.2，没有问题，有-r 参数
> 
> ```
> qrencode version 4.0.2
> Copyright (C) 2006-2017 Kentaro Fukuchi
> Usage: qrencode [OPTION]... [STRING]
> Encode input data in a QR Code and save as a PNG or EPS image.
> 
>   -h           display this message.
>   --help       display the usage of long options.
>   -o FILENAME  write image to FILENAME. If '-' is specified, the result
>                will be output to standard output. If -S is given, structured
>                symbols are written to FILENAME-01.png, FILENAME-02.png, ...
>                (suffix is removed from FILENAME, if specified)
>   -r FILENAME  read input data from FILENAME.
>   -s NUMBER    specify module size in dots (pixels). (default=3)
>   -l {LMQH}    specify error correction level from L (lowest) to H (highest).
>                (default=L)
>   -v NUMBER    specify the minimum version of the symbol. (default=auto)
>   -m NUMBER    specify the width of the margins. (default=4 (2 for Micro))
>   -d NUMBER    specify the DPI of the generated PNG. (default=72)
>   -t {PNG,PNG32,EPS,SVG,XPM,ANSI,ANSI256,ASCII,ASCIIi,UTF8,ANSIUTF8}
>                specify the type of the generated image. (default=PNG)
>   -S           make structured symbols. Version must be specified.
>   -k           assume that the input text contains kanji (shift-jis).
>   -c           encode lower-case alphabet characters in 8-bit mode. (default)
>   -i           ignore case distinctions and use only upper-case characters.
>   -8           encode entire data in 8-bit mode. -k, -c and -i will be ignored.
>   -M           encode in a Micro QR Code.
>   -V           display the version number and copyrights of the qrencode.
>   [STRING]     input data. If it is not specified, data will be taken from
>                standard input.
> 
>   Try "qrencode --help" for more options.
> ```


jdcloudcs
```bash
root@jd-cloud:~# sudo apt-get update -y
sudo: unable to resolve host jd-cloud
Hit:1 http://mirrors.jdcloudcs.com/ubuntu bionic InRelease
Hit:2 http://mirrors.jdcloudcs.com/ubuntu bionic-updates InRelease
Hit:3 http://mirrors.jdcloudcs.com/ubuntu bionic-backports InRelease
Hit:4 http://mirrors.jdcloudcs.com/ubuntu bionic-security InRelease
Reading package lists... Done
root@jd-cloud:~# sudo apt-get install -y qrencode
sudo: unable to resolve host jd-cloud
Reading package lists... Done
Building dependency tree       
Reading state information... Done
qrencode is already the newest version (3.4.4-1build1).
The following package was automatically installed and is no longer required:
  libreadline6
Use 'sudo apt autoremove' to remove it.
0 upgraded, 0 newly installed, 0 to remove and 6 not upgraded.
```

**Comment by sanqiaiziji** [comment]
No such file or directory: './compose/docker-compose.yml'  我尝试使用推荐方式，但是在第一步报错了

**Pull request merged**